src/Microsoft.AspNetCore.Rewrite/UrlRewriteBuilder.cs

```diff
@@ -0,0 +1,167 @@
+        /// <param name="rule">The literal string of a mod_rewrite rule: 
+        /// "RewriteRule Pattern Substitution [Flags]"</param>
+        /// <param name="action">Action to perform on the <see cref="ModRewriteRuleBuilder"/> </param>
+        public static UrlRewriteOptions AddModRule(this UrlRewriteOptions options, string rule, Action<ModRewriteRuleBuilder> action)
+        {
+            var builder = new ModRewriteRuleBuilder(rule);
+            action(builder);
+            AddRule(options, builder.Build());
+
+            return options;
+        }
+
+        /// <summary>
+        /// Adds a mod_rewrite rule to the current rules.
+        /// </summary>
+        /// <param name="options">The UrlRewrite options.</param>
+        /// <param name="rule">The literal string of a mod_rewrite rule: 
+        /// "RewriteRule Pattern Substitution [Flags]"</param>
+        public static UrlRewriteOptions AddModRule(this UrlRewriteOptions options, string rule)
+        {
+            var builder = new ModRewriteRuleBuilder(rule);
+            AddRule(options, builder.Build());
+            return options;
+        }
+
+        public static UrlRewriteOptions RewritePath(this UrlRewriteOptions options, string regex, string newPath, bool stopRewriteOnSuccess = false)
+        {
+            options.Rules.Add(new PathRule { MatchPattern = new Regex(regex), OnMatch = newPath, OnCompletion = stopRewriteOnSuccess ? Transformation.TerminatingRewrite : Transformation.Rewrite });
+            return options;
+        }
+
+        public static UrlRewriteOptions RewriteScheme(this UrlRewriteOptions options, bool stopRewriteOnSuccess = false)
+        {
+            options.Rules.Add(new SchemeRule {OnCompletion = stopRewriteOnSuccess ? Transformation.TerminatingRewrite : Transformation.Rewrite });
+            return options;
+        }
+
+        public static UrlRewriteOptions RedirectPath(this UrlRewriteOptions options, string regex, string newPath, bool stopRewriteOnSuccess = false)
+        {
+            options.Rules.Add(new PathRule { MatchPattern = new Regex(regex), OnMatch = newPath, OnCompletion = Transformation.Redirect });
+            return options;
+        }
+
+        public static UrlRewriteOptions RedirectScheme(this UrlRewriteOptions options, int? sslPort)
+        {
+            options.Rules.Add(new SchemeRule { SSLPort = sslPort, OnCompletion = Transformation.Redirect });
+            return options;
+        }
+
+        public static UrlRewriteOptions CustomRule(this UrlRewriteOptions options, Func<UrlRewriteContext, RuleResult> onApplyRule, Transformation transform, string description = null)
+        {
+            options.Rules.Add(new FunctionalRule { OnApplyRule = onApplyRule, OnCompletion = transform});
+            return options;
+        }
+
+        public static UrlRewriteOptions RulesFromConfig(this UrlRewriteOptions options, IConfiguration rulesFromConfig)
```

Comment:
I think we should also get rid of this until we understand it better. Expressing this generic format in the configuration system is a bit ugly. Do you have an examples of this?

Also, RulesFromConfig should be more verby 😄